### DISCUSSION TITLE
How does grt achieve global routing?

### DISCUSSION THREAD

**Discussion by Icamd** [The-OpenROAD-Project/OpenROAD#3371] [Q&A]
Fastroute is a great global routing algorithm that can complete the routing of millions of two-pin nets in such a short time. However, after reading the Fastroute paper, it does not mention how the program allocates threads for the millions of two-pin nets. Does Fastroute divide all the two-pin nets into multiple portions and assign a thread to each portion? How are these threads synchronized to update the capacity of the whole global routing grid? Thank you!

**Comment by maliberty** [comment]
FR is not multi-threaded.

**Reply by Icamd** [reply]
If Fastroute is not multi-threaded, does it process millions of individual two-pin nets one by one? How does it achieve such incredible processing speed? Thank you

**Reply by maliberty** [reply]
yes serially.  You can read the various papers at https://www.engineering.iastate.edu/~cnchu/FastRoute.html which describe the algorithms.
